node/src/main/kotlin/net/corda/node/internal/NodeStartup.kt

```
@@ -62,7 +63,16 @@ open class NodeStartup(val args: Array<String>) {
 
         drawBanner(versionInfo)
         Node.printBasicNodeInfo(LOGS_CAN_BE_FOUND_IN_STRING, System.getProperty("log-path"))
-        val conf = loadConfigFile(cmdlineOptions)
+        val conf0 = loadConfigFile(cmdlineOptions)
+
+        val conf = if (cmdlineOptions.bootstrapRaftCluster && conf0 is NodeConfigurationImpl) {
```

Should we raise an error if it's not `NodeConfigurationImpl` but they requested bootstrap ?

---

Yes, that's a good idea.

---

I think this would be clearer (excuse the lack of indents!):

`if(cmdlineOptions.bootstrapRaftCluster) {`
`   if(conf0 is NodeConfigurationImpl) {`
`   } else { `
`     // ERROR`
`  }`
`} else {`
`}`